testcases/parser/nodes.go

```diff
@@ -43,6 +45,77 @@ type TestCase struct {
 	FuncType      TestFuncType
 }
 
+func (tc *TestCase) GetFunctionOptions() []*types.FunctionOption {
+	if len(tc.Options) == 0 {
+		return nil
+	}
+	funcOptions := make([]*types.FunctionOption, 0)
+	for key, value := range tc.Options {
+		funcOptions = append(funcOptions, &types.FunctionOption{
+			Name:       key,
+			Preference: []string{value},
+		})
+	}
+	return funcOptions
+}
+
+func (tc *TestCase) scalarSignatureKey() string {
+	var b strings.Builder
+	for i, a := range tc.Args {
+		if i != 0 {
+			b.WriteByte('_')
+		}
+		b.WriteString(a.Type.ShortString())
+	}
+	return b.String()
+}
+
+func (tc *TestCase) aggregateSignatureKey() string {
+	var b strings.Builder
+	for i, a := range tc.AggregateArgs {
+		if i != 0 {
+			b.WriteByte('_')
+		}
+		b.WriteString(a.ColumnType.ShortString())
+	}
+	return b.String()
+}
+
+func (tc *TestCase) signatureKey() string {
+	if tc.FuncType == ScalarFuncType {
+		return tc.scalarSignatureKey()
+	}
+	return tc.aggregateSignatureKey()
+}
+
+func (tc *TestCase) CompoundFunctionName() string {
+	return tc.FuncName + ":" + tc.signatureKey()
+}
+
+func (tc *TestCase) ID() extensions.ID {
+	baseURI := tc.BaseURI
+	if !strings.HasPrefix(baseURI, "https") || !strings.HasPrefix(baseURI, "http") {
```

Comment:
This check is being used to check that the URI is not a slash-prefixed file.  So it could check for the prefix being a leading slash.  Otherwise a check for a leading slash could be useful before appending it onto the Substrait default address.